core/src/network.rs

```diff
@@ -187,7 +187,7 @@ where
     ///
     /// The translation is transport-specific. See [`Transport::address_translation`].
     pub fn address_translation<'a>(&'a self, observed_addr: &'a Multiaddr)
-        -> impl Iterator<Item = Multiaddr> + 'a
+        -> Vec<Multiaddr>
```

Comment:
Is this a necessary change for this PR?  I'm not opposed to returning a `Vec`, but if it is not necessary for this PR I think it would be better to propose it separately.

Reply:
This is to avoid cloning/collecting when calling add_external_addrs when the Report observed address event happens

Reply:
I see, I missed that :+1: .